src/tools/pmreorder/binaryoutputhandler.py

```diff
@@ -43,9 +43,9 @@ def add_file(self, file, map_base, size):
         :type size: int
         :return: None
         """
-        self._files.append(
-            BinaryFile(file, map_base, size, self._checker, self._logger)
-        )
+        binaryfile = BinaryFile(file, map_base, size, self._checker, self._logger)
```

Comment:
You could implement `__contains__` for `BinaryOutputHandler`, and this check would simply become `if file not in self: ...`.
Alternatively, we could create a generalized method to return `BinaryFile` (`lambda x: (x.file_name is file)` ?), and then you could use that both here and in `remove_file`. It would be cleaner, I think.